File: YuGiOh2/Cards/C19613556.cs
using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 大风暴
    /// </summary>
    public class C19613556
    {
        public static int Type { get; } = (int)ChooseTargetType.None;

        public static bool CheckIfAvailable(Player player)
        {
            return player.Enemy.Field.SpellAndTrapFields.Any(c => c != null) ||
                player.Field.SpellAndTrapFields.Any(c => c != null && c.UID != player.EffectingCard.UID);
        }

        public static void ProcessEffect(Player player)
        {
            Destroy(player.Enemy);
            Destroy(player);
        }

        private static void Destroy(Player player)
        {
            for (int i = 0; i < 5; i++)
            {
                if (player.Field.SpellAndTrapFields[i] == null)
                    continue;

                player.AddCardToGrave(player.Field.SpellAndTrapFields[i]);
                player.Field.SpellAndTrapFields[i] = null;
            }
            if (player.Field.FieldField != null)
            {
                player.Enemy.ProcessEnemyField();
            }
            player.TrapsWhenAttack.Clear();
            player.TrapsWhenSummon.Clear();
        }
    }
}

File: YuGiOh2/Scripts/C19613556.lua
Type = 0;
--大风暴
function checkIfAvailable(player)
    for index, spell in pairs(player.Field.SpellAndTrapFields) do
        if spell ~= nil and spell.UID ~= player.EffectingCard.UID then
            return true;
        end
    end
    for index, spell in pairs(player.Enemy.Field.SpellAndTrapFields) do
        if spell ~= nil then
            return true;
        end
    end
    if player.Field.FieldField ~= nil or player.Enemy.Field.FieldField ~= nil then
        return true;
    end
    return false;
end

function processEffect(player)
    destroy(player.Enemy);
    destroy(player);
end

function destroy(player)
    for i, spell in pairs(player.Field.SpellAndTrapFields) do
        if spell ~= nil then
            player:AddCardToGrave(spell);
            player:ClearSpellAndTrapField(i - 1);
        end
    end
    if player.Field.FieldField ~= nil then
        player.Enemy:ProcessEnemyField();
    end
    player:ClearTrapsWhenSummon();
    player:ClearTrapsWhenAttack();
end